dpnp/linalg/linalg_iface.py

```diff
@@ -84,6 +85,61 @@ def eig(in_array1):
     return dpnp_eig(in_array1)
 
 
+def matrix_rank(input, tol=None, hermitian=False):
+    """
+    Return matrix rank of array
+    Rank of the array is the number of singular values of the array that are
+    greater than `tol`.
+
+    Parameters
+    ----------
+    M : {(M,), (..., M, N)} array_like
+        Input vector or stack of matrices.
+    tol : (...) array_like, float, optional
+        Threshold below which SVD values are considered zero. If `tol` is
+        None, and ``S`` is an array with singular values for `M`, and
+        ``eps`` is the epsilon value for datatype of ``S``, then `tol` is
+        set to ``S.max() * max(M.shape) * eps``.
+    hermitian : bool, optional
+        If True, `M` is assumed to be Hermitian (symmetric if real-valued),
+        enabling a more efficient method for finding singular values.
+        Defaults to False.
+    Returns
+    -------
+    rank : (...) array_like
+        Rank of M.
+
+    """
+
+    is_input_dparray = isinstance(input, dparray)
+
+    if not use_origin_backend(input) and is_input_dparray:
+        if tol is not None:
+            checker_throw_value_error("matrix_rank", "tol", type(tol), None)
+        if hermitian is not False:
+            checker_throw_value_error("matrix_rank", "hermitian", hermitian, False)
+
+        result = dpnp_matrix_rank(input)
+
+        # scalar returned
+        if result.shape == (1,):
+            return result.dtype.type(result[0])
+
+        return result
+
+    input1 = dpnp.asnumpy(input) if is_input_dparray else input
+
+    # TODO need to put dparray memory into NumPy call
+    result_numpy = numpy.linalg.matrix_rank(input1, tol=tol, hermitian=hermitian)
+    result = result_numpy
+    if isinstance(result, numpy.ndarray):
+        result = dparray(result_numpy.shape, dtype=result_numpy.dtype)
+        for i in range(result.size):
+            result._setitem_scalar(i, result_numpy.item(i))
+
+    return result
```

Comment:
Please use new method for this.
Example https://github.com/IntelPython/dpnp/blob/11b7f714ea068557ddaadf742fc11eaa5e1df39c/dpnp/dpnp_iface_mathematical.py#L116

Reply:
if it applicable